### PR TITLE
👷 Change build and run analysis conditions

### PR DESCRIPTION
**Pull request by devbyaccident**
This PR makes the following changes to CI:

1. Build and push action will only run if Dockerfile changes are made to dev branch. (More details below)
2. Run Analysis job will run using the latest docker image without rebuilding.

This should allow for an economical use of large Github runners and cut the time to run the analysis job from over 2 and a half hours to about 30 minutes.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Build
+
+on:
+  push:
+    branches:
+      - 'dev'
+    paths:
+      - 'Dockerfile'
```
Comment (devbyaccident) [inline .github/workflows/build.yml:8]: This condition will only trigger if a commit is merged into the `dev` branch **AND** that commit has a change to the *Dockerfile*.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  build_and_publish_image:
+    name: Build and Publish Image
+    runs-on: ubuntu-latest-m
```
Comment (devbyaccident) [inline .github/workflows/build.yml:13]: Using the large runner to build since it will be more infrequent per the condition above.

File: .github/workflows/run_analysis.yml [hunk 3]
```diff
@@ -1,55 +1,12 @@
-name: CI
+name: Analysis
 
 on:
-  push:
-    branches:
-      - dev
   pull_request:
   workflow_dispatch:
```
Comment (devbyaccident) [inline .github/workflows/run_analysis.yml:5]: This is the analysis workflow now. It will trigger due to either of the below conditions:
- A new PR is opened
- A manual workflow is triggered through the GitHub Actions console

File: .github/workflows/run_analysis.yml [hunk 4]
```diff
@@ -166,19 +123,19 @@ jobs:
       - uses: actions/checkout@v3
 
       - name: Download Data
-        uses: docker://pgc-images.sbgenomics.com/d3b-bixu/open-pedcan:analysisjob
+        uses: docker://pgc-images.sbgenomics.com/d3b-bixu/open-pedcan:latest
```
Comment (devbyaccident) [inline .github/workflows/run_analysis.yml:126]: Using the `latest` tag for analysis means that changes to the *Dockerfile* and changes to the packages/data **HAVE** to be separate PRs.

**Review by jharenza — APPROVED** [review]
